plugins/module_utils/podman/podman_pod_lib.py

```diff
@@ -574,9 +573,14 @@ def running(self):
         """Return True if pod is running now."""
         if 'status' in self.info['State']:
             return self.info['State']['status'] == 'Running'
-        if 'status' in self.ps_info:
-            return self.ps_info['status'] == 'Running'
-        return self.info['State'] == 'Running'
+        elif isinstance(self.info['State'],str):
+            return self.info['State'] == 'Running'
+        # older podman versions don't have status in 'podman pod inspect'
+        # if other methods fail, use 'podman pod ps'
+        ps_info = self.get_ps()
+        if 'status' in ps_info:
+            return ps_info['status'] == 'Running'
+        return False
```

Comment:
I don't think we want to change here the default behavior. I prefer it to fail than return incorrect state `False`.

Reply:
Thanks for the help! I've added another commit.